netty/shaded/src/testShadow/java/io/grpc/netty/shaded/ShadingTest.java

```diff
@@ -17,61 +17,96 @@
 package io.grpc.netty.shaded;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
 import io.grpc.Server;
 import io.grpc.ServerBuilder;
 import io.grpc.internal.testing.TestUtils;
 import io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts;
 import io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder;
+import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContextBuilder;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslProvider;
 import io.grpc.stub.StreamObserver;
 import io.grpc.testing.protobuf.SimpleRequest;
 import io.grpc.testing.protobuf.SimpleResponse;
 import io.grpc.testing.protobuf.SimpleServiceGrpc;
 import io.grpc.testing.protobuf.SimpleServiceGrpc.SimpleServiceBlockingStub;
 import io.grpc.testing.protobuf.SimpleServiceGrpc.SimpleServiceImplBase;
-import org.junit.Before;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** Unit tests for {@link Shading}. */
 @RunWith(JUnit4.class)
 public final class ShadingTest {
+  private ManagedChannel channel;
+  private Server server;
+
+  @After
+  public void tearDown() throws Exception {
+    if (channel != null) {
+      channel.shutdownNow();
+      channel.awaitTermination(1, TimeUnit.SECONDS);
+    }
+    if (server != null) {
+      server.shutdownNow();
+      server.awaitTermination(1, TimeUnit.SECONDS);
+    }
+  }
+
   /** Verify that normal Netty didn't leak into the test runtime. */
   @Test(expected = ClassNotFoundException.class)
   public void noNormalNetty() throws Exception {
     Class.forName("io.grpc.netty.NettyServerBuilder");
   }
 
-  /** Test TLS since that verifies everything, including tcnative, is functional. */
   @Test
-  public void tls() throws Exception {
-    Server server = ServerBuilder.forPort(0)
+  public void serviceLoaderFindsNetty() throws Exception {
+    assertTrue(ServerBuilder.forPort(0) instanceof NettyServerBuilder);
```

Comment:
optional: if you use `Truth.assertThat(ServerBuilder.forPort(0)).isInstanceOf(NettyServerBuilder.class)`, a test failure will tell you what the instance actually was.

Reply:
I swapped the tests to use Truth exclusively.